pkg/services/ngalert/database.go

```diff
@@ -0,0 +1,120 @@
+func (ng *AlertNG) DeleteAlertDefinitionByID(query *DeleteAlertDefinitionByIDQuery) error {
+	return ng.SQLStore.WithTransactionalDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		res, err := sess.Exec("DELETE FROM alert_definition WHERE id = ?", query.ID)
+		if err != nil {
+			return err
+		}
+
+		rowsAffected, err := res.RowsAffected()
+		if err != nil {
+			return err
+		}
+		query.RowsAffected = rowsAffected
+		return nil
+	})
+}
+
+// GetAlertDefinitionByID is a handler for retrieving an alert definition from that database by its ID.
+// It returns models.ErrAlertDefinitionNotFound if no alert definition is found for the provided ID.
+func (ng *AlertNG) GetAlertDefinitionByID(query *GetAlertDefinitionByIDQuery) error {
+	return ng.SQLStore.WithTransactionalDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		alertDefinition, err := getAlertDefinitionByID(query.ID, sess)
+		if err != nil {
+			return err
+		}
+		query.Result = alertDefinition
+		return nil
+	})
+}
+
+// SaveAlertDefinition is a handler for saving a new alert definition.
+func (ng *AlertNG) SaveAlertDefinition(cmd *SaveAlertDefinitionCommand) error {
+	return ng.SQLStore.WithTransactionalDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		alertDefinition := &AlertDefinition{
+			OrgId:     cmd.OrgID,
+			Name:      cmd.Name,
+			Condition: cmd.Condition.RefID,
+			Data:      cmd.Condition.QueriesAndExpressions,
+		}
+
+		if err := ng.validateAlertDefinition(alertDefinition, cmd.SignedInUser, cmd.SkipCache); err != nil {
+			return err
+		}
+
+		if err := alertDefinition.preSave(); err != nil {
+			return err
+		}
+
+		if _, err := sess.Insert(alertDefinition); err != nil {
+			return err
+		}
+
+		cmd.Result = alertDefinition
+		return nil
+	})
+}
+
+// UpdateAlertDefinition is a handler for updating an existing alert definition.
+// It returns models.ErrAlertDefinitionNotFound if no alert definition is found for the provided ID.
+func (ng *AlertNG) UpdateAlertDefinition(cmd *UpdateAlertDefinitionCommand) error {
+	return ng.SQLStore.WithTransactionalDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		alertDefinition := &AlertDefinition{
+			Name:      cmd.Name,
+			Condition: cmd.Condition.RefID,
+			Data:      cmd.Condition.QueriesAndExpressions,
+		}
+
+		if err := ng.validateAlertDefinition(alertDefinition, cmd.SignedInUser, cmd.SkipCache); err != nil {
+			return err
+		}
+
+		if err := alertDefinition.preSave(); err != nil {
+			return err
+		}
+
+		affectedRows, err := sess.ID(cmd.ID).Update(alertDefinition)
+		if err != nil {
+			return err
+		}
+
+		cmd.Result = alertDefinition
+		cmd.RowsAffected = affectedRows
+		return nil
+	})
+}
+
+// GetAlertDefinitions is a handler for retrieving alert definitions of specific organisation.
+func (ng *AlertNG) GetAlertDefinitions(cmd *ListAlertDefinitionsCommand) error {
```

Comment:
Doesn't have to be public:
```suggestion
// getAlertDefinitions is a handler for retrieving alert definitions of specific organisation.
func (ng *AlertNG) getAlertDefinitions(cmd *ListAlertDefinitionsCommand) error {
```